### ISSUE TITLE
Release pycparser v2.21

### ISSUE THREAD

**Issue by eliben** [eliben/pycparser#438]
The previous release (2.20) was quite a while ago (Mar 2020). Lots of changes has accumulated since.

@vit9696 are you planning additional significant C11 work any time soon?

I'd be happy to get help testing before a new release is made.

**Comment by vit9696** [comment]
Hi @eliben, I actually wanted to ask you for a release myself :)

To be fair, I wanted to perform a few more tests of pycparser on a certain codebase before that happens in case we missed something non-trivial. I am also relatively concerned about https://github.com/eliben/pycparser/issues/433 as it may cause very serious with generation, which we rely on to generate function wrappers for testing, but at this step I do not have a good suggestion. It will be nice if you can wait by the end of the week, and release then if I keep silent.

In general I have no plans to add new features soon. The remaining C11 features that would possibly require pycparser updates are:

- _Generic
- Anonymous strctures
- Unicode string literals
- Complex number macros

Feel free to mention anything that I missed. I do not think I personally need any of these, but it would be nice to have full C11 compatibility in pycparser eventually in the future.

**Comment by eliben** [comment]
I'm fine with waiting, there's no rush to release whatsoever. The additional testing you're mentioning is precisely the kind of vetting I would hope will happen before I make a new release.

This is also a call to everyone who's only running/testing their stuff with a released version -- please run/test it with the tip of master branch and report any serious issues, and in particular regressions.

**Comment by vit9696** [comment]
Our stuff seems to be working fine, but I actually ended up adding unicode string support just in case we need it in https://github.com/eliben/pycparser/pull/439. Anonymous structures already are supported by pycparser, and as for _Imaginary_I or _Complex_I these would rather need preprocessor stuff than actual pycparser support.

This leaves us just with the `_Generic`s, but I am really not into it. It is messy and has little use in real world from my own perspective. I guess you can update the README to say that C18 is mostly supported aside `_Generic` with some features from C23.

**Comment by eliben** [comment]
v2.21 has been release to PyPI: https://pypi.org/project/pycparser/2.21/
And tagged on GitHub.

Marking the issue fixed; please reopen if you observe problems.

**Comment by johnernberg** [comment]
The added argument align to the c_ast Decl object constructor breaks the API between 2.20 and 2.21, was that intended?

**Comment by eliben** [comment]
> The added argument align to the c_ast Decl object constructor breaks the API between 2.20 and 2.21, was that intended?

Can you clarify in what way the API is broken, with an example? It'd be best to open a new issue for this.

Generally, though, it's hard to add features without adding fields on the AST nodes.
